client/src/components/Header/components/language-list.tsx

```diff
@@ -0,0 +1,203 @@
+  };
+
+  const handleMenuKeyDown = (
+    event: React.KeyboardEvent<HTMLButtonElement | HTMLAnchorElement>
+  ) => {
+    if (event.key === 'Escape') {
+      listButtonRef.current?.focus();
+      setShowList(false);
+      event.preventDefault();
+    }
+  };
+
+  const handleBlur = (event: React.FocusEvent<HTMLButtonElement>) => {
+    if (
+      event.relatedTarget &&
+      !event.relatedTarget.closest('.nav-list') &&
+      event.relatedTarget !== listButtonRef.current
+    ) {
+      setShowList(false);
+    }
+  };
+
+  const handleLastLangaugeKeys = (
+    event: React.KeyboardEvent<HTMLButtonElement | HTMLAnchorElement>
+  ) => {
+    const DoKeyPress = new Map<string, { select: () => void }>([
+      [
+        'Escape',
+        {
+          select: () => {
+            listButtonRef.current?.focus();
+            setShowList(false);
+            event.preventDefault();
+          }
+        }
+      ],
+      [
+        'Tab',
+        {
+          select: () => {
+            const camperPressedTheShiftKey = event.shiftKey;
+            if (!camperPressedTheShiftKey) {
+              setShowList(false);
+            }
+          }
+        }
+      ]
+    ]);
+    DoKeyPress.get(event.key)?.select();
+  };
+
+  const handleFirstLangaugeKeys = (
+    event: React.KeyboardEvent<HTMLButtonElement | HTMLAnchorElement>
+  ) => {
+    const DoKeyPress = new Map<string, { select: () => void }>([
+      [
+        'Escape',
+        {
+          select: () => {
+            listButtonRef.current?.focus();
+            setShowList(false);
+            event.preventDefault();
+          }
+        }
+      ],
+      [
+        'Tab',
+        {
+          select: () => {
+            const camperPressedTheShiftKey = event.shiftKey;
+            if (camperPressedTheShiftKey) {
+              setShowList(false);
+            }
+          }
+        }
+      ]
+    ]);
+    DoKeyPress.get(event.key)?.select();
+  };
+
+  const getHandleLanguageKeys = (languagePosition: number) => {
+    const lastLanguage = locales.length - 1;
+
+    if (languagePosition === 0) {
+      return handleFirstLangaugeKeys;
+    } else if (languagePosition === lastLanguage) {
+      return handleLastLangaugeKeys;
+    } else handleMenuKeyDown;
+  };
+
+  return pending ? (
+    <div className='nav-skeleton' />
+  ) : (
+    <>
+      <button
+        aria-controls='nav-lang-list'
+        aria-expanded='true'
+        className='lang-button-nav'
+        onClick={handleClick}
+        onBlur={handleBlur}
+        onKeyDown={handleMenuKeyDown}
+        title={t('buttons.change-language')}
+        aria-label={t('buttons.change-language')}
+        ref={listButtonRef}
+      >
+        <LanguageGlobe />
+      </button>
+      <ul
+        aria-labelledby='toggle-button-nav'
+        id='nav-lang-list'
+        className={`nav-list${showList ? ' nav-lang-list' : ''} `}
+      >
+        {locales.map((lang, index) => (
+          <li key={'lang-' + lang} role='none'>
```

Comment:
```suggestion
          <li key={'lang-' + lang}>
```

We need to get rid of this role too.